### PR TITLE
56 (take 2) chunks of youtube transcripts should include their timestamp offsets 

### LINKED ISSUE #56: Chunks of YouTube transcripts should include their timestamp offsets
### Overview

Currently, we get and store the offset (in seconds) with each bit of text. But when we form chunks of text to pass to an LLM, we discard the offset.

### Requirements

Track the offset of each chunk.

### Notes
If the chunks are long, the offset might be quite a long way before the claims within the chunk.

### PR DESCRIPTION
**Pull request by dcorney**
Fixes #56 .

**Stores the start & end offset for each chunk of a transcript**

DON'T MERGE YET! The corresponding change needs to be made to the backend `app.py` as claims now have fields `offset_s` and `offset_end_s`. 

Do we want to define these in seconds or milliseconds? (Seconds seems fine-grained enough to me but open to other opinions).

And either way, we need to update the database schema to store offset end.

-----------------

## Pull request checklist

- [x] I have linked my PR to an issue
- [x] I’ve used [conventional commits](https://github.com/FullFact/automation-docs/wiki/Conventional-commits)
- [x] My branch is up-to-date with `main`
- [ ] Where appropriate, I have added or updated tests
- [ ] Where appropriate, I have [updated documentation](https://github.com/FullFact/automation-docs/) to reflect my changes

### CODE REVIEW

File: src/health_misinfo_shared/youtube_api.py [hunk 1]
```diff
@@ -104,18 +104,32 @@ def load_texts(folder) -> list[dict]:
...
+    """Split/merged a list of sentences into series of overlapping text chunks.
+    Each chunk is a dict containing the text and the start/end timestamps"""
     current_chunk_text = ""
-    for sentence in transcript:
-        current_chunk_text += sentence["sentence_text"] + " "
-        if len(current_chunk_text) > 5000:
-            yield current_chunk_text
+    current_chunk_start_offset = 0.0
+    current_chunk_end_offset = 0.0
+    for s in transcript_obj:
```
Comment (andylolz) [inline src/health_misinfo_shared/youtube_api.py:113]: This is minor, but I think `sentence` is a clearer variable name than `s`.

File: src/health_misinfo_shared/youtube_api.py [hunk 2]
```diff
@@ -104,18 +104,32 @@ def load_texts(folder) -> list[dict]:
...
+        current_chunk_text += s["sentence_text"] + " "
+        if len(current_chunk_text) > 1500:
```
Comment (andylolz) [inline src/health_misinfo_shared/youtube_api.py:115]: Just noting this change to chunk size. Is this deliberate? If the LLM is happy with this, then that’s great – I suspect these smaller chunks will work much better for the offsets.

**Review by andylolz — CHANGES_REQUESTED** [review]
A couple of small changes requested (typing and variable naming) but overall this looks great.

I don’t think this PR is blocked by API changes – I think this PR can be merged and everything will continue to work, and the API changes can happen as a separate step.

**Comment by andylolz** [comment]
> Do we want to define these in seconds or milliseconds? (Seconds seems fine-grained enough to me but open to other opinions).

I agree – but switching to seconds would mean renaming the database field!

File: src/health_misinfo_shared/youtube_api.py (hunk 1 above)
Comment (dcorney) [reply src/health_misinfo_shared/youtube_api.py:113]: Agreed. I'll fix.

**Review by andylolz — APPROVED** [review]
Looks good! Thanks for the fixes

**Comment by andylolz** [comment]
I realised you were right all along, @dcorney – the seconds to milliseconds change did introduce a breaking change here.

I’ve made a very small fix for it (which will presumably be replaced with something better soon) and I think this is now mergeable as is.

**Pull request merged**